### PR TITLE
[NEXUS-8311] Hook up info/jmx.json to support-zip

### PR DESCRIPTION
**Pull request by jdillon**
https://issues.sonatype.org/browse/NEXUS-8311

Includes change which re-enables local JMX connections, at the expense of loosing RBAC for the time being until KARAF-3147 is resolved or we re-implement RBAC control for JMX.

### CODE REVIEW

**Comment by jdillon** [comment]
If `management` feature is not installed, platform mbean server pukes up:

```
java.lang.IllegalStateException: KarafMBeanServerBuilder not initialized
```

If `management` is installed, requests fail due to lack of auth.

Both are unexpected for accessing the local MBeanServer IMO, but Karaf is doing something here which prevents typical in-JVM operations.

**Comment by jdillon** [comment]
http://bamboo.s/browse/NX3-OSSF140-1 :white_check_mark:

**Comment by mcculls** [comment]
+1 though I think we should put markers in our customized scripts documenting what we've changed to make it easier to upgrade to later versions of Karaf.

**Comment by jdillon** [comment]
@mcculls whats sorta markers you thinking of?

**Comment by mcculls** [comment]
basically just a comment before any modified lines, explaining what changed and why - not essential, but can help when it comes to merge conflicts (esp. since anyone updating these files when we update karaf will probably be doing a manual merge)

**Comment by jdillon** [comment]
@mcculls I've added some "SONATYPE: ..." markers for future ref.  Noticed that we haven't fully changed all the scripts for our changes, like for example bin/client still has the -Dcom.sun.management.jmxremote properties, etc.

Anyways, should be slightly better now, lemme know if there were more changes that needed markers.

File: plugins/internal/nexus-atlas-plugin/src/main/java/org/sonatype/nexus/atlas/internal/customizers/JmxCustomizer.groovy [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+          }
+
+          model[name] = attrs
+        }
+
+        file.text = new JsonBuilder(model).toPrettyString()
+      }
+
+      def assignable(type, types) {
+        for (from in types) {
```
Comment (kellyrob99) [inline plugins/internal/nexus-atlas-plugin/src/main/java/org/sonatype/nexus/atlas/internal/customizers/JmxCustomizer.groovy]: looks like a good candidate for any:
```
return types.any{from -> from.isAssignableFrom(type)}
```

File: plugins/internal/nexus-atlas-plugin/src/main/java/org/sonatype/nexus/atlas/internal/customizers/JmxCustomizer.groovy [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        }
+
+        // TODO: Cope with password-like fields where we can detect .*password.* or something?
+
+        def type = value.getClass()
+        log.trace "Rendering type: $type"
+
+        if (value instanceof TabularData) {
+          def result = []
+          value.keySet().each {
```
Comment (kellyrob99) [inline plugins/internal/nexus-atlas-plugin/src/main/java/org/sonatype/nexus/atlas/internal/customizers/JmxCustomizer.groovy]: collect probably suits this use case better

**Comment by kellyrob99** [comment]
+1

**Comment by cstamas** [comment]
+1

**Comment by jdillon** [comment]
manually merged

**Pull request closed without merging**